### PR TITLE
Add vyos_bgp_global resource module

### PR DESCRIPTION
**Pull request by GomathiselviS**
##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/vyos_bgp_global.py [hunk 1]
```diff
@@ -0,0 +1,1433 @@
...
+The module file for vyos_bgp_global
+"""
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: vyos_bgp_global
+version_added: 1.3.0
```
Comment (NilashishC) [inline plugins/modules/vyos_bgp_global.py:35]: Should this be 2.0.0?

File: plugins/modules/vyos_bgp_global.py [hunk 2]
```diff
@@ -0,0 +1,1433 @@
...
+short_description: BGP Global Resource Module.
+description:
+- This module manages BGP global configuration of interfaces on devices running VYOS.
+author: Gomathi Selvi Srinivasan (@GomathiselviS)
+options:
+  config:
+    description: A dict of BGP global configuration for interfaces.
+    type: dict
+    suboptions:
+      as_num:
```
Comment (NilashishC) [inline plugins/modules/vyos_bgp_global.py:45]:
```suggestion
      as_number:
```

File: plugins/modules/vyos_bgp_global.py [hunk 3]
```diff
@@ -0,0 +1,1433 @@
...
+        description:
+        - AS number.
+        type: int
+        required: True
+      aggregate_address:
+        description:
+          - BGP aggregate network.
+        type: dict
+        suboptions:
+          address:
```
Comment (NilashishC) [inline plugins/modules/vyos_bgp_global.py:55]: Can we name this prefix? Looks this the expected values are in CIDR format.

File: plugins/modules/vyos_bgp_global.py [hunk 4]
```diff
@@ -0,0 +1,1433 @@
...
+      description:
+      - The state the configuration should be left in.
+      - State I(purged) removes all the BGP configurations from the
+        target device. Use caution with this state.('no router bgp <x>')
+      - State I(deleted) only removes BGP attributes that this modules
+        manages and does not negate the BGP process completely. Thereby, preserving
+        address-family related configurations under BGP context.
+      - Running states I(deleted) and I(replaced) will result in an error if there
+        are address-family configuration lines present under vrf context that is
+        is to be removed. Please use the  M(arista.eos.eos_bgp_address_family)
```
Comment (NilashishC) [inline plugins/modules/vyos_bgp_global.py:413]:
```suggestion
        is to be removed. Please use the  M(vyos.vyos.vyos_bgp_address_family)
```

File: plugins/modules/vyos_bgp_global.py [hunk 5]
```diff
@@ -0,0 +1,1433 @@
...
+        executing the command B(show running-config | section bgp).
+      - The state I(parsed) reads the configuration from C(running_config) option and
+        transforms it into Ansible structured data as per the resource module's argspec
+        and the value is then returned in the I(parsed) key within the result.
+      type: str
+    state:
+      description:
+      - The state the configuration should be left in.
+      - State I(purged) removes all the BGP configurations from the
+        target device. Use caution with this state.('no router bgp <x>')
```
Comment (NilashishC) [inline plugins/modules/vyos_bgp_global.py:407]:
```suggestion
        target device. Use caution with this state.('delete protocols bgp <x>')
```

File: plugins/modules/vyos_bgp_global.py [hunk 6]
```diff
@@ -0,0 +1,1433 @@
...
+      - State I(deleted) only removes BGP attributes that this modules
+        manages and does not negate the BGP process completely. Thereby, preserving
+        address-family related configurations under BGP context.
+      - Running states I(deleted) and I(replaced) will result in an error if there
+        are address-family configuration lines present under vrf context that is
```
Comment (NilashishC) [inline plugins/modules/vyos_bgp_global.py:412]: Is vrf supported in VyOS?

File: plugins/modules/vyos_bgp_global.py [hunk 7]
```diff
@@ -0,0 +1,1433 @@
...
+            description: BGP router-id
+            type: str
+          scan_time:
+            description: BGP route scanner interval
+            type: int
+    running_config:
+      description:
+      - This option is used only with state I(parsed).
+      - The value of this option should be the output received from the EOS device by
+        executing the command B(show running-config | section bgp).
```
Comment (NilashishC) [inline plugins/modules/vyos_bgp_global.py:398]:
```suggestion
        executing the command B(show configuration commands |  match "set protocols bgp").
```

File: plugins/modules/vyos_bgp_global.py [hunk 8]
```diff
@@ -0,0 +1,1433 @@
...
+          router_id:
+            description: BGP router-id
+            type: str
+          scan_time:
+            description: BGP route scanner interval
+            type: int
+    running_config:
+      description:
+      - This option is used only with state I(parsed).
+      - The value of this option should be the output received from the EOS device by
```
Comment (NilashishC) [inline plugins/modules/vyos_bgp_global.py:397]:
```suggestion
      - The value of this option should be the output received from the VyOS device by
```

File: tox.ini [hunk 9]
```diff
@@ -26,7 +26,7 @@ commands = {posargs}
 # E123, E125 skipped as they are invalid PEP-8.
 
 show-source = True
-ignore = E123,E125,E402,W503
+ignore = E123,E125,E402,W503,E501
```
Comment (NilashishC) [inline tox.ini:29]: Could you please add a note in the file for why we need to ignore E501?

**Review by rohitthakur2590 — APPROVED** [review]

**Comment by GomathiselviS** [comment]
recheck

**Pull request merged**